src/gui/tray/svgimageprovider.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) by Oleksandr Zolotov <alex@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include <QQuickImageProvider>
+
+namespace OCC {
+namespace Ui {
+    class SvgImageProvider : public QQuickImageProvider
```

Comment:
@mgallien There is no network request, just reading and rending the local file, so, I am assuming it should be fast enough, given that `createSvgPixmapWithCustomColor` is using `QPixmapCache`, or, not? FYI, there is already one `ImagProvider` within `usermodel.h` which is also rendering local SVG image, and it was there for a while without any sign of UI being blocked because of that, should we reconsider and use async image provider then? I wonder, why it would be needed as we are not doing network fetching in both providers.

Reply:
that could be fine but the whole point of using an async provider is that the generation of the image can be offloaded to another thread instead of keeping the main thread busy
so @allexzander, you think that is not needed here ?

Reply:
@mgallien Isn't there additional overhead when the new thread is created? In that case, we could use a thread pool. But, this needs to be implemented. I am not convinced this is needed in this case. We are not reading from the network socket and we are caching the resource. The overhead of using the main thread is minimal. If you insist, I will still implement the QThreadPool logic.

Reply:
@allexzander from my understand qml will manage the threads. you don't need to start any threads or thread pools inside the async image provider.

Reply:
@FlexW In Qt's example they write quite a lot of code to manage threads https://code.qt.io/cgit/qt/qtdeclarative.git/tree/examples/quick/imageresponseprovider/imageresponseprovider.cpp?h=5.15

Reply:
Okay then I may be wrong, but are you sure that is not to just demonstrate what you could do inside the async image provider? Because when I read https://doc.qt.io/qt-5/qquickasyncimageprovider.html#requestImageResponse it says that this function will be called by multiple threads. Normally you have just one GUI thread and the GUI thread then starts workers. Because of that, I assumed that the one who called `requestImageResponse()` will already be a worker thread.

Reply:
@FlexW Because they are using `QThreadPool` as a member, it doesn't seem like this can be avoided.